es-app/src/SystemScreenSaver.cpp

```diff
@@ -138,59 +218,49 @@ void SystemScreenSaver::renderScreenSaver()
 			mVideoScreensaver->render(transform);
 		}
 	}
-	else if (mState != STATE_INACTIVE)
+	else if (mImageScreensaver && Settings::getInstance()->getString("ScreenSaverBehavior") == "slideshow")
```

Comment:
Preferably assign a variable to ScreenSaverBehaviour setting and then compare rather than multiple fetching.